### PR TITLE
Scandinavian language layouts

### PR DESCRIPTION
**Pull request by Merkost**

### CODE REVIEW

**Comment by Merkost** [comment]
#162

**Comment by tibbi** [comment]
check this conflict too please

**Comment by tibbi** [comment]
languages were sorted alphabetically before, why have you split them this way? Even 3 english variants are far from each other now

**Comment by Merkost** [comment]
> languages were sorted alphabetically before, why have you split them this way? Even 3 english variants are far from each other now, everything is random now.

Implemented in Language.sorted function. #173

**Comment by tibbi** [comment]
but what is the sense of adding them so randomly in the enum itself? Makes no sense to me and it makes maintaining the app harder

File: app/src/main/kotlin/com/simplemobiletools/keyboard/helpers/Constants.kt [hunk 1]
```diff
@@ -21,20 +25,72 @@ const val SHOW_NUMBERS_ROW = "show_numbers_row"
...
-const val LANGUAGE_SPANISH = 4
-const val LANGUAGE_GERMAN = 5
-const val LANGUAGE_ENGLISH_DVORAK = 6
-const val LANGUAGE_ROMANIAN = 7
-const val LANGUAGE_SLOVENIAN = 8
-const val LANGUAGE_BULGARIAN = 9
-const val LANGUAGE_TURKISH_Q = 10
-const val LANGUAGE_LITHUANIAN = 11
-const val LANGUAGE_BENGALI = 12
-const val LANGUAGE_GREEK = 13
```
Comment (Merkost) [inline app/src/main/kotlin/com/simplemobiletools/keyboard/helpers/Constants.kt:91]: I have added languages in enum exactly how they were written in constants

File: app/src/main/kotlin/com/simplemobiletools/keyboard/extensions/Context.kt [hunk 2]
```diff
@@ -149,40 +149,9 @@ fun Context.setupKeyboardDialogStuff(
...
-        RadioItem(LANGUAGE_LITHUANIAN, getKeyboardLanguageText(LANGUAGE_LITHUANIAN)),
-        RadioItem(LANGUAGE_ROMANIAN, getKeyboardLanguageText(LANGUAGE_ROMANIAN)),
-        RadioItem(LANGUAGE_RUSSIAN, getKeyboardLanguageText(LANGUAGE_RUSSIAN)),
-        RadioItem(LANGUAGE_SLOVENIAN, getKeyboardLanguageText(LANGUAGE_SLOVENIAN)),
-        RadioItem(LANGUAGE_SPANISH, getKeyboardLanguageText(LANGUAGE_SPANISH)),
-        RadioItem(LANGUAGE_TURKISH_Q, getKeyboardLanguageText(LANGUAGE_TURKISH_Q)),
-    )
+    val languages = Language.sorted(this).map { RadioItem(it.ordinal, it.getName(this)) }
+    return ArrayList(languages)
 }
```
Comment (Merkost) [inline app/src/main/kotlin/com/simplemobiletools/keyboard/extensions/Context.kt:154]: The languages were sorted by hardcode the radioItems in this function before.

File: app/src/main/kotlin/com/simplemobiletools/keyboard/extensions/Context.kt [hunk 3]
```diff
@@ -149,40 +149,9 @@ fun Context.setupKeyboardDialogStuff(
...
-        RadioItem(LANGUAGE_FRENCH, getKeyboardLanguageText(LANGUAGE_FRENCH)),
-        RadioItem(LANGUAGE_GERMAN, getKeyboardLanguageText(LANGUAGE_GERMAN)),
-        RadioItem(LANGUAGE_GREEK, getKeyboardLanguageText(LANGUAGE_GREEK)),
-        RadioItem(LANGUAGE_LITHUANIAN, getKeyboardLanguageText(LANGUAGE_LITHUANIAN)),
-        RadioItem(LANGUAGE_ROMANIAN, getKeyboardLanguageText(LANGUAGE_ROMANIAN)),
-        RadioItem(LANGUAGE_RUSSIAN, getKeyboardLanguageText(LANGUAGE_RUSSIAN)),
-        RadioItem(LANGUAGE_SLOVENIAN, getKeyboardLanguageText(LANGUAGE_SLOVENIAN)),
-        RadioItem(LANGUAGE_SPANISH, getKeyboardLanguageText(LANGUAGE_SPANISH)),
-        RadioItem(LANGUAGE_TURKISH_Q, getKeyboardLanguageText(LANGUAGE_TURKISH_Q)),
-    )
```
Comment (Merkost) [inline app/src/main/kotlin/com/simplemobiletools/keyboard/extensions/Context.kt:167]: The languages were sorted by hardcode the radioItems in this function before.

Comment (Merkost) [reply app/src/main/kotlin/com/simplemobiletools/keyboard/extensions/Context.kt:167]: I have added the sort by the string value of the language (its name), which seems more properly to me than just sorting it by its English name.

**Comment by tibbi** [comment]
lets make numbers selected by default at long pressing the top row, in danish it is not the case at 8 and 9

**Comment by tibbi** [comment]
and format xml files with ctrl + alt + L, there are a few misplaces brackets

**Comment by tibbi** [comment]
yeah should be good now, thanks

**Pull request merged**